core/classes/Core/Util.php

```diff
@@ -143,26 +218,21 @@ public static function getSelfURL(bool $protocol = true): string {
         }
 
         // https and www checks
-        if (self::isConnectionSSL()) {
-            $proto = 'https://';
-        } else {
-            $proto = 'http://';
-        }
+        $proto = self::getProtocol() . '://';
```

Comment:
Might read easier if we change the param from `protocol` to `show_protocol`, then we can rename `proto` to `protocol`